Directory.Packages.props

```diff
@@ -25,6 +25,9 @@
     <PackageVersion Include="Refit" Version="7.1.2" />
     <PackageVersion Include="System.Data.SqlClient" Version="4.8.6" />
     <PackageVersion Include="System.Linq.Async" Version="6.0.1" />
+    <PackageVersion Include="ApprovalTests" Version="6.0.0" />
```

Comment:
Sjekka Githuben til disse og ser de anbefaler å bruke Verify istedenfor: https://github.com/approvals/ApprovalTests.Net/blob/master/readme.md

Hva bruker vi dette til egentlig?

Reply:
Jeg bruker dette til å hente ut pathen til testene slik at jeg lagrer brukte tokens på riktig sted, og at dette blir riktig på tvers av OS og forskjellige måter å kjøre testene på (IDE, terminal, etc).

Jeg visste at dette test biblioteket hadde løst dette problemet før, og det var ikke et problem jeg hadde lyst til å løse på tvers av utvikler maskiner hos oss, så valgte å bruke deres implementasjon (se "IntegrarionTest" base klassa der jeg finner pathen).

Er også derfor jeg ikke bruker verify biblioteket :)

Reply:
Hvis det er mulig med Verify burde vi bruke det, siden denne pakka blir ikke vedlikeholdt.
Kommer også opp en warning her
![image](https://github.com/user-attachments/assets/ed223482-135b-4a30-b9f1-119d84c79a2c)

Reply:
Kan sikkert switche den ja.

Er ikke så redd for vulnerabilities i en test pakke som aldri skal nå produksjon og som ingenting annet avhenger av, men kan bytte ut :)

Reply:
Ut ifra det du skriver:
> lagrer brukte tokens på riktig sted, og at dette blir riktig på tvers av OS og forskjellige måter å kjøre testene på (IDE, terminal, etc).

Burde ikke det være noe som er ganske godt støttet i .NET uten at vi trenger å dra inn tredjepartskomponenter? Hva er det som gjør at det som er i .NET ikke er godt nok? Jeg ville tro at System.Io.File dekker godt nok det behovet vi har her, med mindre det er noe jeg ikke ser 😄

Reply:
Grunnen til at jeg valgte å dra inn noe er fordi at jeg ville ha noe som er godt testet på forskjellige scenarioer. Kjører man dotnet test på i toppmappen? I prosjekt mappen? Via CLI? Vil forskjeller i hvordan man har satt opp aliaser i kommandolinja gjøre en forskjell? Via VS? Via vim (som jeg gjør)? I wsl? I windows? Hva om man får noen med macos? etc.

Jeg ønsker at filene skal havne på samme sted uansett hvordan man gjør det. Litt for at ting skal være consistent, men også for at da blir det plukket opp av .gitignore.

Kan man gjøre dette med vanlig .NET? Sure. De som lagde ApprovalTests klarte det med vanlig .NET. Men jeg har gjort lignende før hvor forskjeller på utviklermaskinene har ført til at en og en ting blir plukket opp og må tas hensyn til. Så da valgte jeg å strekke meg ut til noe som er mer battle tested :)

Problemet er ikke 'System.Io.File', men å finne riktig path på tvers av lokale konfigurasjoner en utvikler kan ha lokalt.

Reply:
Vil tro at microsoft sin [System.Runtime.CompilerServices.CallerFilePath](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callerfilepathattribute?view=net-8.0) dekker det du nevner her.
Den får tak i stien til den som har kalt på tjenesten.

Du kan se på koden i personoppslag.api for å se hvordan vi har gjort det tidligere.
`Setupfixture.cs`
```
    public static void AddInfo(string fnr,
    [System.Runtime.CompilerServices.CallerMemberName] string memberName = "",
    [System.Runtime.CompilerServices.CallerFilePath] string sourceFilePath = "")
#pragma warning restore NUnit1028 // The non-test method is public
    {
        var className = sourceFilePath.Split("\\").Last().Split(".").First();
        var name = $"{className}.{memberName}";
        if (!TestInfo.ContainsKey(name))
        {
            TestInfo.Add(name, [fnr]);
        }
        else
        {
            var ptr = TestInfo.GetValueOrDefault(name);
            ptr.Add(fnr);
        }
    }
```

Reply:
Takker @yne-bv. Dependency fjernet.